code/controllers/subsystem/processing/quirks.dm

```diff
@@ -36,14 +39,18 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 		quirk_points[initial(T.name)] = initial(T.value)
 
 /datum/controller/subsystem/processing/quirks/proc/AssignQuirks(mob/living/user, client/cli, spawn_effects)
-	var/badquirk = FALSE
+	var/bad_quirk_checker = 0
 	for(var/V in cli.prefs.active_character.all_quirks)
 		var/datum/quirk/Q = quirks[V]
 		if(Q)
 			user.add_quirk(Q, spawn_effects)
+			bad_quirk_checker += initial(Q.value)
 		else
-			stack_trace("Invalid quirk \"[V]\" in client [cli.ckey] preferences")
-			cli.prefs.active_character.all_quirks -= V
-			badquirk = TRUE
-	if(badquirk)
+			stack_trace("Invalid quirk \"[V]\" in client [cli.ckey] preferences. the game has reset their quirks automatically.")
+			bad_quirk_checker = BAD_QUIRK_FLAG
+	if(bad_quirk_checker) // negative & zero value = calculation good / positive quirk value = something's wrong
+		cli.prefs.active_character.all_quirks = list()
 		cli.prefs.active_character.save(cli)
+		client_alert(cli, "You have an outdated quirk, or more of them. Your eligible quirks are kept at this round, but your character preference has been reset. Please re-apply your quirks.", "Oh, no!")
```

Comment:
Just have it say
`Your selected character quirk \"[V]\" is no longer present in the game. Your selected quirks have been reset, please review them at any time.`

Reply:
The failsafe covers both case of removal or quirk point adjustment rather than handling specific case. Your correction isn't how it works.

Reply:
This failsafe is to cover both of two cases - quirk removal and quirk point adjustment. your correction is not appliable to quirk point adjustment because you don't know which one was adjusted.
So, this failsafe is made to cover things generally.

Reply:
It should say the quirk name still so they know, also it should say "one or more invalid quirks"

Reply:
changed